src/pdn/src/via.cpp

```diff
@@ -1322,22 +1334,42 @@ bool ViaGenerator::appliesToLayers(odb::dbTechLayer* lower,
 
 void ViaGenerator::getMinimumEnclosures(std::vector<Enclosure>& bottom, std::vector<Enclosure>& top, bool rules_only) const
 {
-  auto populate_enc = [this](odb::dbTechLayer* layer, int width, bool above, std::vector<Enclosure>& encs) {
+  auto populate_enc = [this](odb::dbTechLayer* layer,
+                             int width,
+                             odb::dbTechLayerDir direction,
+                             bool above,
+                             std::vector<Enclosure>& encs) {
     for (auto* rule : getCutMinimumEnclosureRules(width, above)) {
-      encs.emplace_back(rule, layer, getCut());
+      encs.emplace_back(rule, layer, getCut(), direction);
     }
   };
 
   populate_enc(getBottomLayer(),
                getLowerWidth(false),
+               getRectDirection(lower_rect_),
                false,
                bottom);
   populate_enc(getTopLayer(),
                getUpperWidth(false),
+               getRectDirection(upper_rect_),
                true,
                top);
 }
 
+odb::dbTechLayerDir ViaGenerator::getRectDirection(const odb::Rect& rect) const
+{
+  const int height = rect.dy();
+  const int width = rect.dx();
+
+  if (width < height) {
+    return odb::dbTechLayerDir::VERTICAL;
+  }
+  if (height < width) {
+    return odb::dbTechLayerDir::HORIZONTAL;
+  }
+  return odb::dbTechLayerDir::NONE;
+}
+
```

Comment:
Very similar to `Rect::getDir()`

Reply:
Yes, and I would use it if it returned the enum, instead of -1, 1, and 0 which makes the code hopeless to read

Reply:
I agree, maybe the odb function should be replaced by something like this?

Reply:
I think it would make sense, either replace or add something like `getDirection()` (which could use the `getDir()` or vice versa.) also if the enum should be reserved for techlayer, it could use another one or make one. I only used the `dbTechLayerDir` one since the names are easy to read.

Reply:
@maliberty this seems like a good use case for [`Orientation2D`](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/odb/include/odb/isotropy.h#L98), but there is no `None` value there.